ui/src/project.ts

```diff
@@ -321,6 +493,56 @@ export const repositoryPathValidationStore = (
   }
 };
 
+const VALID_PEER_MATCH = /^[\w\d]{54}$/;
+
+const checkPeerUniqueness = (peer: string): Promise<boolean> => {
```

Comment:
Just so you know, track and untrack are idempotent so they won't double track or anything. But if this is for UX then totaly ignore.

Reply:
Thanks, yeah, was aware of that. It's only for UX.